### ISSUE TITLE
How to run an excluded test?

### ISSUE THREAD

**Issue by atreeon** [dart-lang/test#1764]
I have decorated my test file with `@Tags(['integration'])` at the top
I then created a `dart_test.yaml` file with `exclude_tags: integration`
https://github.com/dart-lang/test/blob/master/pkgs/test/doc/configuration.md#include_tags

If I run all my tests my integration test doesn't run - as expected.

How do I run that test?  I don't want to accidentally run other integration tests so think commenting out the tags or dart_test.yaml code would be error prone.

**Comment by jakemac53** [comment]
You need to pass that tag explicitly `-t integration`. If you want to only run that integration test, make sure to also pass the path to that test explicitly. So `dart test -t integration test/my_integration_test.dart`.

**Comment by atreeon** [comment]
I'm still not able to get this to override the dart_test.yaml exclude.

My dart_test.yaml:
```exclude_tags: integration```

My command
```dart test -t integration test/blah_test.dart```

Tag at top of test
```@Tags(['integration'])```

Output with the dart_test.yaml exclude tags specified
```No tests ran.
No tests match the requested tag selectors:
  include: "integration"
  exclude: "integration"```

Output without tag
```00:01 +1: All tests passed!  ```

Maybe the test is being excluded and excluded and the exclude overrides the include???  I want to override the exclude but maybe that is not possible???

Apologies @jakemac53 for delay, I was on an older version of Dart and I wanted to upgrade.
Dart - version 2.18.2
Test - 1.21.6

**Comment by jakemac53** [comment]
Hmm yeah, it is possible the excludes override the includes. The precedence gets pretty weird here, because probably what you really want is for _command line includes_ to not be overridden. And then having overlapping includes/excludes in the dart.yml file would be an error? I don't know that we really have good infra for supporting that right now, and all the merging of these args is already quite complicated as it stands...

**Comment by jakemac53** [comment]
cc @natebosch thoughts?

**Comment by atreeon** [comment]
yup, it seems like it doesn't do what I want it to do.  Maybe we need a exclude_tags_unless_specified item in the test_dart.yaml ???  I guess I could create a saved test for each project that excludes those 'integration' tests but it would be nice to do it on a project level (or even global).

**Comment by jakemac53** [comment]
Maybe you could also try setting up two different [presets](https://github.com/dart-lang/test/blob/master/pkgs/test/doc/configuration.md#presets) instead to work around this. So the exclude config wouldn't exist at all in the other preset.

**Comment by natebosch** [comment]
Yes, I think the design intention here was that you'd create two presets.

I do think it would be reasonable for `-t` command line arguments to take precedence, but I don't know how complicated that would be to implement.

**Comment by atreeon** [comment]
My pain is not too severe thankfully but I think it has at times stopped me writing some testing which is possibly a problem.

Thinking some more now with the extra info you've kindly provided, it might be more of the interface between IDEA and the test commands where my specific problem is.

Idea uses the following commands:
Right click the test & run
dart run test -r json myprojectdir/test/calculator_test.dart -N "subtraction"

Right clicking on a file & run:
dart run test -r json myprojectdir/test/calculator_test.dart

Right clicking on the project & running all my tests uses the command
dart run test -r json myprojectdir

I use this right clicking and running my tests a lot while I'm developing.

When my 'integration' tag is on the individual test and I run the test file or test suit as a whole, I want that test to be excluded but if I ran the individual test I want that specific test to run.

Similarly if the tag is on the individual file and I ran the test file I would expect it to run but excluded from when I run all my tests.

I can set some presets up and I think they will get me some way there and I'll get by but I think the above is kind of sensible.  Saying that it is the interface may not be exactly true as I'd probably want / expect the same from the command too.
